types/test.tsx

```diff
@@ -35,13 +36,18 @@ export async function testPureRender() {
 
   // helpers
   const {container, rerender, debug} = page
+  expectType<HTMLElement, typeof container>(container)
   return {container, rerender, debug}
 }
 
 export function testRenderOptions() {
   const container = document.createElement('div')
   const options = {container}
-  render(<div />, options)
+  const {container: returnedContainer} = render(<div />, options)
+  // Unclear why TypeScript infers `HTMLElement` here when the the input `container` is `HTMLDivElement`
+  // Hopefully this breaks someday and we can switch to
+  // expectType<HTMLDivElement, typeof returnedContainer>(returnedContainer)
+  expectType<HTMLElement, typeof returnedContainer>(returnedContainer)
```

Comment:
This is a limitation of TypeScript's JSX syntax support. JSX elements unfortunately can't be inferred to a specific element type. You'd have to use the `React.createElement` syntax, though it's not recommended in most cases.

So for now, this test is accurate, though you could avoid JSX if you wanted to still test the regression.

Reply:
The `<div />` is irrelevant here as far as I know though definitely confusing. The type should be inferred from `options`. 

The confusing part is why it works for `SVGSVGElement` but not `HTMLDivElement`. 

Let me replace `<div />` with `<button />` to check if your assumption is true.

Reply:
Hm, I missed that, I'm curious what happens

Reply:
Changed. Is it now clearer that the `container` is not inferred from the rendered UI but the `options`?